### DISCUSSION TITLE
Failing to identify subpopulation diversity

### DISCUSSION THREAD

**Discussion by asmagen** [GreenleafLab/ArchR#818] [Questions / Documentation]
I've been analyzing various mouse immune cell 10x scATACseq datasets and found rich diversity of myeloid cells using ArchR clustering, but so far the same strategy fails when applied to human immune cells although the sequencing/data quality was good -- I was wondering if you can point me to strategies/ArchR functionality to identify the source of the problem.
The specific issue is that myeloid subsets (dendritic cells, macrophages and monocytes) don't appear to segregate into distinct clusters as I saw before in other contexts.
This is the relevant code to my initial clustering strategy:
```
human_TN <- addIterativeLSI(
        ArchRProj = human_TN,
        useMatrix = "TileMatrix", 
        name = "IterativeLSI", 
        iterations = 4, 
        clusterParams = list( #See Seurat::FindClusters
            resolution = c(0.1, 0.2, 0.4),
            sampleCells = 10000, 
            n.start = 10
        ), 
        varFeatures = 25000, 
        dimsToUse = 1:30
    )

    human_TN <- addClusters(
        input = human_TN,
        reducedDims = "IterativeLSI",
        method = "Seurat",
        name = "Clusters",
        resolution = 0.8,
        force = T
    )
```
which I then refined by performing peak calling and using the peak matrix for feature selection, which didn't help by much.

Based on the top differentially accessible genes by gene score I do find some of the relevant markers popping up such as C1QC for macrophages, but it's non specific and seems to be mixed with dendritic cell markers like CD1C.
Some shared features might be driving them together like IL1B which is a known inflammatory feature, but also a lot of MIR/LINC genes pop up everywhere and I'm not sure if they confound the clustering and results?
Top differential genes: <img width="643" alt="Screen Shot 2021-06-10 at 3 43 41 PM" src="https://user-images.githubusercontent.com/11299568/121587315-2bf45b80-ca24-11eb-80a7-abf52b1f9dd6.png">
Selected marker genes: <img width="566" alt="Screen Shot 2021-06-10 at 3 48 17 PM" src="https://user-images.githubusercontent.com/11299568/121587906-d0769d80-ca24-11eb-80eb-7062684dd7ac.png">

How do I interrogate this problem? Whether the features relevant to distinguishing these cell types? How do I extract the features used for clustering and annotate them by closest gene? Any other strategies to identify the source of this issue?

Thank you!

Cc'ing my colleague @s7hegde

**Comment by rcorces** [comment]
Please note that the Issues section is meant for Bug reports, which this is not.
![image](https://user-images.githubusercontent.com/10343046/121628992-2cdeba80-ca2f-11eb-8813-767e8191adf3.png)

I will migrate your issue to the Discussions section where it belongs.

**Comment by rcorces** [comment]
We arent able to comment on user-specific analyses. If we did this, we would have a lot of similar requests. While we hope to improve the documentation in the future, the manual represents our advice as best as we can.

Our hope is that one day other members of the community participate in the discussions posted here to help each other.

**Comment by asmagen** [comment]
Okay, so focusing on non user specific analysis, how do you propose interrogating the features (bins/peaks) selected for clustering in order to QC and debug issues? Surely it's a basic mechanism that every user is required to do if they want to understand their data.

**Reply by rcorces** [reply]
You could try to extract this information from the SVD matrix (stored in the `reducedDims` object). Though it seems like the most straightforward method is to select the groups of cells that you are interested in and look for what is differential between them using `getMarkerFeatures()`. The underlying problem is that you are interested in why you arent observing distinct clusters which is far more challenging to understand and fix than if you were trying to understand why two clusters are different.
